src/pages/iou/MoneyRequestCreatedPage.js

```diff
@@ -37,44 +47,29 @@ const defaultProps = {
         comment: '',
         created: '',
         participants: [],
+        receiptPath: '',
     },
 };
 
-class MoneyRequestCreatedPage extends Component {
-    constructor(props) {
-        super(props);
+function MoneyRequestCreatedPage({iou, route}) {
+    const {translate} = useLocalize();
+    const iouType = lodashGet(route, 'params.iouType', '');
+    const reportID = lodashGet(route, 'params.reportID', '');
 
-        this.updateDate = this.updateDate.bind(this);
-        this.navigateBack = this.navigateBack.bind(this);
-        this.iouType = lodashGet(props.route, 'params.iouType', '');
-        this.reportID = lodashGet(props.route, 'params.reportID', '');
-    }
-
-    componentDidMount() {
-        const moneyRequestId = `${this.iouType}${this.reportID}`;
-        const shouldReset = this.props.iou.id !== moneyRequestId;
+    useEffect(() => {
+        const moneyRequestId = `${iouType}${reportID}`;
+        const shouldReset = iou.id !== moneyRequestId;
         if (shouldReset) {
             IOU.resetMoneyRequestInfo(moneyRequestId);
         }
 
-        if (_.isEmpty(this.props.iou.participants) || (this.props.iou.amount === 0 && !this.props.iou.receiptPath) || shouldReset) {
-            Navigation.goBack(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID), true);
+        if (_.isEmpty(iou.participants) || (iou.amount === 0 && !iou.receiptPath) || shouldReset) {
+            Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType, reportID), true);
         }
-    }
+    }, [iou.id, iou.participants, iou.amount, iou.receiptPath, iouType, reportID]);
```

Comment:
we want to run this only on component mount only, right?

we'll need to use an empty dependency array then.

Reply:
actually, iou could change if we edit on another device

Reply:
So I can leave it like this, right?

Reply:
yes, looks equivalent to the class based implementation.